lib/fastlane/plugin/aws_s3/actions/aws_s3_action.rb

```diff
@@ -97,6 +99,10 @@ def self.run(config)
           Aws::S3::Client.new
         end
 
+        if obtain_path_from_gym
```

Comment:
What I mean is:

```
if xcarchive_file == "obtain_path_from_gym"
  xcarchive_file = Actions.lane_context[SharedValues::XCODEBUILD_ARCHIVE]
end
```

So we get rid of the "obtain_path_from_gym" parameter, and you can do:

`xcarchive = "obtain_path_from_gym"`

Or:

`xcarchive = "/any/path/you/want/file.xcarchive"`

Or not specify `xcarchive` at all, so the archive is not uploaded